uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/ArcGISAuthenticationController.cpp

```diff
@@ -138,28 +140,70 @@ void ArcGISAuthenticationController::handleArcGISAuthenticationChallenge(ArcGISA
   emit displayUsernamePasswordSignInView();
 }
 
+/*!
+  \internal
+ */
+void ArcGISAuthenticationController::handleNetworkAuthenticationChallenge(NetworkAuthenticationChallenge* challenge)
+{
+  std::lock_guard<std::mutex> lock(m_mutex);
+  challenge->setParent(nullptr);
+  m_currentNetworkChallenge = std::unique_ptr<NetworkAuthenticationChallenge>{challenge};
+
+  if (!m_currentNetworkChallenge)
+  {
+    return;
+  }
+
+  if (!m_currentNetworkChallenge->sslErrors().isEmpty())
```

Comment:
We will need a better way to tell network challenges apart. That will come later once we have more of them.